### PR TITLE
SelectEdgeByProjectedCoordinate

### PR DESCRIPTION
**Pull request by mastep84**
Added function 'SelectEdgeByProjectedCoordinate' selecting the edge for which the specified coordinate is closest to its perpendicular projection onto this edge.

### CODE REVIEW

**Comment by sreiter** [comment]
What you're caclulating is not necessarily the distance to the projected point. If the projection lies outside of a line segment, the distance to the closest endpoint of the line segment (edge) is computed. The naming is somewhat misleading and probably unnecessarily complex, I guess.

Why not simply `SelectClosestEdge`?

File: tools/selection_tools.cpp [hunk 1]
```diff
@@ -606,6 +606,39 @@ void SelectSubsetEdgesByDirection(
 	                       maxDeviationAngle, selectFlipped);
 }
 
+Edge* SelectEdgeByMinPerpendicularDistance(
```
Comment (sreiter) [inline tools/selection_tools.cpp]: As commented somewhere, this is not accurate - if the projection of the given coordinate on an edge lies outside of that edge, the distance to the closest endpoint is considered.

File: tools/selection_tools.cpp [hunk 2]
```diff
@@ -606,6 +606,39 @@ void SelectSubsetEdgesByDirection(
...
+			const vector3& coord)
+{
+	Grid& grid = m->grid();
+	Mesh::position_accessor_t& aaPos = m->position_accessor();
+
+	if(grid.begin<Edge>() == grid.end<Edge>())
+		return NULL;
+
+	EdgeIterator iter = grid.begin<Edge>();
+	Edge* bestEdge = *iter;
```
Comment (sreiter) [inline tools/selection_tools.cpp]: maybe initialize with NULL and add a check below inside the loop:
```
if (bestEdge == NULL || dist < bestDist)
{...}
```
You could then use a for loop and avoid code duplication. You could also remove the empty-check above.

**Comment by mastep84** [comment]
Dear Sebastian,

thank you for your review -- indeed, the naming was somewhat misleading. I have applied modifications according to your suggestions streamlining the code.

Best regards,
Martin

**Comment by sreiter** [comment]
Dear Martin,

I'm sorry that it took so long to merge your request. Thanks for taking the time to consider my comments :)

Best,
Sebastian

**Pull request merged**